R/rw.R

```diff
@@ -0,0 +1,229 @@
+#' Reweighting factor container
+#'
+#' This function `rw()` creates containers for reweighting factors
+#' of class `rw`. This class is particularly designed to store reweighting 
+#' factors for each gauge configuration, that can be applied on 
+#' correlation functions emerging in statistical and quantum field theory
+#' simulations. Arithmetic operations are defined for this class in
+#' several ways, as well as for increasing statistics and \link{is.rw}.
+#'
+#' @details
+#'#'
+#' @family rw constructors
+#'
+#' @export
+rw <- function () {
+  rw <- list()
+  class(rw) <- append(class(rw), 'rw')
+  return (rw)
+}
+
+#' rw metadata mixin constructor
+#'
+#' @param .rw `rw` object to extend.
+#' @param conf.index list of Integers, containing the index of gauge configurations
+#'
+#' @family cf constructors
+#'
+#' @export
+rw_meta <- function (.rw = rw(), conf.index = NA) {
+   stopifnot(inherits(.rw, 'rw'))
+
+   .rw$conf.index = conf.index
+
+   class(.rw) <- append(class(.rw), 'rw_meta')
+   return (.rw)
+}
+
+
+#' @export
+rw_orig <- function (.rw = rw(), rw, conf.index, max_value) {
+  stopifnot(inherits(.rw, 'rw'))
+
+  .rw$rw <- rw
+  .rw$max_value <- max_value
+  .rw$conf.index <- conf.index
+
+  class(.rw) <- append(class(.rw), 'rw_orig')
+  class(.rw) <- append(class(.rw), 'rw_meta')
+
+  ## Norm is needed in order to be able to extend statistics
+  ## If the object does not have `rw_norm` it cannot be extended
+  class(.rw) <- append(class(.rw), 'rw_norm')
+  return (.rw)
+}
+
+#' @export
+rw_unit <- function (.rw = rw(), conf.index) {
+  stopifnot(inherits(.rw, 'rw'))
+
+  .rw$conf.index=conf.index
+  .rw$rw <- rep(1.0,length(conf.index))
+  .rw$max_value <- 1.0
+
+  class(.rw) <- append(class(.rw), 'rw_orig')
```

Comment:
the constructors should do this for you, no?

Reply:
yes, it should